### ISSUE TITLE
Add tutorial using just Lightning Module 

### ISSUE THREAD

**Issue by kordc** [SebChw/Actually-Robust-Training#163]

**Comment by SebChw** [comment]
It'll be added as a quickstart to readme

**Comment by SebChw** [comment]
I`ll create a test from the quickstart

**Comment by SebChw** [comment]
Here is my initial draft implementation please comment on this


```python
import math

import torch.nn as nn
from torchmetrics import Accuracy
from utils.quickstart_modules import YourLightningDataModule, YourLightningModule

from art.checks import CheckScoreCloseTo, CheckScoreLessThan
from art.metrics import MetricCalculator, SkippedMetric
from art.project import ArtProject
from art.steps import CheckLossOnInit, Overfit, OverfitOneBatch


def test_quickstart1():
    """Simplest possible usage of ART"""

    datamodule = YourLightningDataModule()
    model_class = YourLightningModule

    project = ArtProject(datamodule)

    # For this to work we should allow providing empty checks. And just print warning... man use checks.
    project.add_step(CheckLossOnInit(model_class))
    project.add_step(OverfitOneBatch(model_class))
    project.add_step(Overfit(model_class))

    # For this to work we must omit any calls to metric_calculator or set_metric_calculator
    # I still support idea of Metric calculator but it should be explicitly governed by the user. In the leter phase of the development.
    project.run_all()
    project.print_summary()


def test_quickstart2():
    """Adding checks to ART"""

    datamodule = YourLightningDataModule()
    model_class = YourLightningModule

    project = ArtProject(datamodule)

    # Let's assume we work on typical classification_problem
    NUM_CLASSES = 10
    EXPECTED_LOSS = -math.log(1 / NUM_CLASSES)

    # Here we must allow providing custom metric name
    project.add_step(
        CheckLossOnInit(model_class),
        checks=[CheckScoreCloseTo("logged_loss_name", EXPECTED_LOSS, rel_tol=0.001)],
    )
    project.add_step(
        OverfitOneBatch(model_class),
        checks=[CheckScoreLessThan("logged_loss_name", 0.001)],
    )
    project.add_step(
        Overfit(model_class),
        checks=[CheckScoreLessThan("Some_training_metric_name_here", 0.001)],
    )

    project.run_all()
    project.print_summary()


def test_quickstart3():
    """Start utilizing MetricCalculator"""
    datamodule = YourLightningDataModule()
    model_class = YourLightningModule

    """Here goes modified model_class.train_set that uses metric calculator"""

    # Let's assume we work on typical classification_problem
    NUM_CLASSES = 10
    EXPECTED_LOSS = -math.log(1 / NUM_CLASSES)

    # User governs MetricCalculator by themselves (do this explicitly)
    metric = Accuracy(task="multiclass", num_classes=NUM_CLASSES)
    loss_fn = nn.CrossEntropyLoss()
    metric_calculator = MetricCalculator([metric, loss_fn])
    project = ArtProject(datamodule, metric_calculator=metric_calculator)

    project.add_step(
        CheckLossOnInit(model_class),
        checks=[CheckScoreCloseTo(loss_fn, EXPECTED_LOSS, rel_tol=0.001)],
        skipped_metrics=[SkippedMetric(metric=metric)],
    )
    project.add_step(
        OverfitOneBatch(model_class),
        checks=[CheckScoreLessThan(metric, 0.001)],
    )
    project.add_step(
        Overfit(model_class),
        checks=[CheckScoreLessThan(metric, 0.001)],
    )

    project.run_all()
    project.print_summary()

```

**Comment by kordc** [comment]
I'll go over three quickstarts:
1. 
* Agree, we could just add a simple `if no check` and provide an empty check printing results and informing the user about the good of using checks.
* Okay, it'll be slightly more work to omit all calls, as for now we fully depend on it. But I agree it'll be beneficial
2. I think we already allow providing custom metrics, don't we? Maybe not the name, but if someone defines the loss_fn also in the main he can pass it to the check. I think it could work. It'd be easier than specifying the exact name
3. I don't see any difference in the use of a Metric Calculator between your code and the tutorial ones. But I understand it's just connected with the 1. problem, that metric calculator needs to be optional, not obligatory. 

I think those changes won't break the ART's idea, but we have to really carefully do these changes. I also support the usage of the metric calculator. 


One more thought - If we do the quickstart3 should we stay at this loss and everything calculation with those dicts with input, predictions etc.? Maybe we could get rid of them and rely on the lightning's way of doing things. I don't see any usecase but the one that loss calculation using metric calculator is the same for every model. But maybe one would like to train models with different losses, and the metric calculator will be then a tool to have the same metrics just for model comparison in terms of metrics?

**Comment by SebChw** [comment]
2. Without using metric calculator we have no idea under what name user has logged their metrics. So they must know the name
3. The main difference is that user now is explicitly aware of something like MetricCalculator. And now they **must use it by themselves** it won't be automated.

Basically all of this is to allow user move smoothly to using the ArtModule. 

About this one more thought - I don't understand this. The general idea of ArtModule was to systematize stuff.

I'll try to reiterate with your issues in mind and send next proposal.
